lib/storage/index_db.go

```diff
@@ -456,8 +466,9 @@ func mustUnmarshalMetricIDs(dst []uint64, src []byte) []uint64 {
 	if err != nil {
 		logger.Panicf("FATAL: cannot decompress metricIDs: %s", err)
 	}
-	if len(dstBuf) == dstBufLen {
-		// Zero metricIDs
+	if len(dstBuf) == dstBufLen+1 && dstBuf[len(dstBuf)-1] == 0 {
```

Comment:
This check should be moved to the beginning of `mustUnmarshalMetricIDs()` function:

```go
if len(src) == 1 && src[0] == 0 {
  // There is an empty metricIDs list.
  return dst
}
```